Userland/Utilities/tail.cpp

```diff
@@ -84,10 +84,10 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
     args_parser.set_general_help("Print the end ('tail') of a file.");
     args_parser.add_option(follow, "Output data as it is written to the file", "follow", 'f');
     args_parser.add_option(line_count, "Fetch the specified number of lines", "lines", 'n', "number");
-    args_parser.add_positional_argument(file, "File path", "file");
+    args_parser.add_positional_argument(file, "File path", "file", Core::ArgsParser::Required::No);
     args_parser.parse(arguments);
 
-    auto f = TRY(Core::File::open(file, Core::OpenMode::ReadOnly));
+    auto f = TRY(Core::File::open((file == "-"sv || file == nullptr) ? "/dev/stdin" : file, Core::OpenMode::ReadOnly));
```

Comment:
We have `File::open_file_or_standard_stream` for that logic.

Reply:
I suppose, although then the whole thing would need to be adapted to use a `Stream::File` instead of just a `File`

Reply:
> [...] would need to be adapted to use a `Stream::File` [...]

The only issue there would be how to implement waiting for more input data, I don't think `Stream::File` has an equivalent for `can_read` yet. The remaining required changes only appear to be nicer error handling and usage of buffers everywhere.

Reply:
>  I don't think `Stream::File` has an equivalent for `can_read` yet.
I'm not super familiar with this project, but I wasn't able to find any method that supported that kind of behavior.